### PR TITLE
Stream

### PR DESCRIPTION
**Pull request by shligit**
Add a simple policy for NVMe write time life hint

### CODE REVIEW

File: include/rocksdb/env.h [hunk 1]
```diff
@@ -152,6 +152,15 @@ class Env {
                                      unique_ptr<RandomAccessFile>* result,
                                      const EnvOptions& options)
                                      = 0;
+  // These values match Linux definition
```
Comment (sagar0) [inline include/rocksdb/env.h]: Could you also add a link here to either linux kernel code or documentation mentioning lifetimes like this?

File: db/column_family.cc [hunk 2]
```diff
@@ -995,6 +995,24 @@ Status ColumnFamilyData::SetOptions(
...
+  if (initial_cf_options_.compaction_style != kCompactionStyleLevel) {
+    return Env::WLTH_NOT_SET;
+  }
+  if (level == 0 || level == 1) {
+    return Env::WLTH_SHORT;
+  }
+  int base_level = current_->storage_info()->base_level();
+
+  // L1: short, L2: medium, ...
+  if (level - base_level >= 3) {
```
Comment (sagar0) [inline db/column_family.cc]: Where do you set medium for level-2 ?

Comment (shligit) [reply db/column_family.cc]: that's the: 
level - base_level +
+                            static_cast<int>(Env::WLTH_SHORT)

File: include/rocksdb/env.h (hunk 1 above)
Comment (shligit) [reply include/rocksdb/env.h]: I'll add a link

File: env/io_posix.cc [hunk 3]
```diff
@@ -858,6 +863,17 @@ bool PosixWritableFile::IsSyncThreadSafe() const { return true; }
 
 uint64_t PosixWritableFile::GetFileSize() { return filesize_; }
 
+void PosixWritableFile::SetWriteLifeTimeHint(Env::WriteLifeTimeHint hint) {
+#ifdef OS_LINUX
+  if (hint == Env::WLTH_NOT_SET && write_hint_ == Env::WLTH_NOT_SET) {
```
Comment (ajkr) [inline env/io_posix.cc]: would `if (write_hint_ == hint) { return; }` work?

**Comment by ajkr** [comment]
BTW, do you plan to share benchmark results here or in a FB post?

**Comment by sagar0** [comment]
Lets also update the summary and description to give everyone an idea about why we are adding this 😄

**Comment by shligit** [comment]
I'm writing notes, hopefully post it out in FB group tomorrow.

**Comment by UncP** [comment]
In your test, the WAF = total_nand_write / total_user_write;
the `total_user_write` is how many we put in the db, what is `total_nand_write`?

**Pull request closed without merging**